homeassistant/components/tradfri/base_class.py

```diff
@@ -61,10 +61,13 @@ def device_info(self):
 
         return {
             "identifiers": {(TRADFRI_DOMAIN, self._device.id)},
-            "name": self._name,
             "manufacturer": info.manufacturer,
             "model": info.model_number,
+            "name": self._name,
+            "power_source": info.power_source_str,
+            "serial": info.serial,
             "sw_version": info.firmware_version,
+            "battery_level": info.battery_level,
             "via_device": (TRADFRI_DOMAIN, self._gateway_id),
         }
 
```

Comment:
Are there any constants available for these keys or are they named on a per-integration-basis?

Reply:
I don't think we have constants.

The device info keys are defined by the core. Only certain existing keys can be used.

Reply:
These are the only constants we have:
https://github.com/home-assistant/home-assistant/blob/476f24e451ee1797f7c67aed080da5a8178eca97/homeassistant/helpers/device_registry.py#L28-L30